src/main/scala/nl/biopet/sbtbiopet/BiopetPlugin.scala

```diff
@@ -0,0 +1,147 @@
+package nl.biopet.sbtbiopet
+
+import sbt._
+import Keys._
+import com.typesafe.sbt.site.SitePlugin
+import com.typesafe.sbt.site.SitePlugin.autoImport.{
+  siteDirectory,
+  siteSubdirName,
+  makeSite
+}
+import com.typesafe.sbt.site.laika.LaikaSitePlugin
+import com.typesafe.sbt.site.laika.LaikaSitePlugin.autoImport.LaikaSite
+import com.typesafe.sbt.site.SiteScaladocPlugin
+import com.typesafe.sbt.site.SiteScaladocPlugin.autoImport.SiteScaladoc
+import com.typesafe.sbt.sbtghpages.GhpagesPlugin
+import com.typesafe.sbt.sbtghpages.GhpagesPlugin.autoImport.{
+  ghpagesRepository,
+  ghpagesCleanSite,
+  ghpagesPushSite
+}
+import laika.sbt.LaikaSbtPlugin.LaikaKeys.{Laika, rawContent}
+import sbtassembly.AssemblyPlugin
+import sbtassembly.AssemblyPlugin.autoImport.assembly
+import com.typesafe.sbt.SbtGit.git
+import com.typesafe.sbt.SbtPgp
+import sbtrelease.ReleasePlugin
+import ReleasePlugin.autoImport.ReleaseTransformations._
+import ReleasePlugin.autoImport.{
+  ReleaseStep,
+  releaseProcess,
+  releaseStepCommand
+}
+
+object BiopetPlugin extends AutoPlugin {
+  override val requires: Plugins = {
+    SitePlugin &&
+    GhpagesPlugin &&
+    AssemblyPlugin &&
+    SbtPgp &&
+    ReleasePlugin &&
+    LaikaSitePlugin &&
+    SiteScaladocPlugin
+  }
+  override lazy val globalSettings: Seq[Setting[_]] = BiopetGlobalSettings
+  override lazy val projectSettings: Seq[Setting[_]] = BiopetProjectSettings
+  object autoImport extends BiopetKeys
+  import autoImport._
+
+  def BiopetGlobalSettings: Seq[Setting[_]] = {
+
+    Seq(
+      // Publication to nexus repository
+      resolvers += Resolver.sonatypeRepo("snapshots"),
+      publishTo := {
```

Comment:
This can be replaced for:
```scala
publishTo := Some(
  if (isSnapshot.value)
    Opts.resolver.sonatypeSnapshots
  else
    Opts.resolver.sonatypeStaging
)
```

This is less error prone